### DISCUSSION TITLE
Switching virtual workspace with MX Master 3

### DISCUSSION THREAD

**Discussion by sna-scourtney** [PixlOne/logiops#449] [Show and tell]
Virtual desktops are an important tool for me, and I need a fast way to switch between them. With the Logitech MX Master 3, I've found that the two left-side buttons below the sideways scroll wheel are great for this. These are by default mapped to "Forward" and "Backward", but I'd rather use them for workspaces.

With Logiops, I'm using the following in /etc/logid.cfg to accomplish this function:

```
devices: (
{
	name: "Wireless Mouse MX Master 3"
	dpi: 1000;
	buttons: (
		{
			cid: 0x53;
			action =
			{
				type: "Keypress";
				keys: ["KEY_LEFTALT", "KEY_LEFTCTRL", "KEY_LEFT"];
			};
		},
		{
			cid: 0x56;
			action =
			{
				type: "Keypress";
				keys: ["KEY_LEFTALT", "KEY_LEFTCTRL", "KEY_RIGHT"];
			};
		}
	);
}
```

The resolution and SmartShift settings are just copied from the example config. The button section is simple but took some experimentation to find which of the several "Forward" and "Back" control IDs were used on this mouse, which is why I'm sharing this working config.

The keys can be anything you want; I picked the combination of left-ctrl, left-alt, and the left or right arrow, because these aren't used in my other apps,

Once the key mapping is set up in Logiops, I used the Actions setting in Cinnamon Desktop to assign those keystrokes to workspace right (for Forward) and workspace left (for Backward). Other Linux desktop environments should have equivalent keyboard shortcut features, but I'm using Cinnamon so that's what I can document. :)

Thanks so much for making and maintaining Logiops. I've preferred Linux since I discovered it in 1999 but have been stuck in Windows for the past several years and am finally able to move back. I'd gotten "spoiled" using these two buttons with the Logitech Options tool in Windows, and I really missed this timesaving shortcut.

**Comment by sly4** [comment]
I had the exact same challenge.  Logiops was the only solution I found that would work.  I could not get xvinput, solaar, or ratbag to take the KEY_LEFTMETA as a keypress mapped to the gesture button.  As I'm on pop_os, I could not sort out how to change the required key combination and struggled with other mouse config options.

THANK YOU!
